pilot/pkg/networking/core/v1alpha3/listener_waypoint.go

```diff
@@ -271,7 +271,7 @@ func (lb *ListenerBuilder) buildWaypointInternal(wls []WorkloadAndServices, svcs
 		}
 	}
 
-	{
+	if len(wls) > 0 {
 		// Direct pod access chain.
 		directChain := &listener.FilterChain{
```

Comment:
can we keep the chain  and just never match to it

Reply:
This is not a right fix. I think it's CIDR validation that gets triggered on empty set.